lib/ruby_lsp/requests/folding_ranges.rb

```diff
@@ -18,276 +18,154 @@ module Requests
     class FoldingRanges < BaseRequest
       extend T::Sig
 
-      SIMPLE_FOLDABLES = T.let(
-        [
-          SyntaxTree::ArrayLiteral,
-          SyntaxTree::BlockNode,
-          SyntaxTree::Case,
-          SyntaxTree::ClassDeclaration,
-          SyntaxTree::For,
-          SyntaxTree::HashLiteral,
-          SyntaxTree::Heredoc,
-          SyntaxTree::ModuleDeclaration,
-          SyntaxTree::SClass,
-          SyntaxTree::UnlessNode,
-          SyntaxTree::UntilNode,
-          SyntaxTree::WhileNode,
-          SyntaxTree::Else,
-          SyntaxTree::Ensure,
-          SyntaxTree::Begin,
-        ].freeze,
-        T::Array[T.class_of(SyntaxTree::Node)],
-      )
-
-      NODES_WITH_STATEMENTS = T.let(
-        [
-          SyntaxTree::IfNode,
-          SyntaxTree::Elsif,
-          SyntaxTree::In,
-          SyntaxTree::Rescue,
-          SyntaxTree::When,
-        ].freeze,
-        T::Array[T.class_of(SyntaxTree::Node)],
-      )
-
-      StatementNode = T.type_alias do
-        T.any(
-          SyntaxTree::IfNode,
-          SyntaxTree::Elsif,
-          SyntaxTree::In,
-          SyntaxTree::Rescue,
-          SyntaxTree::When,
-        )
-      end
-
       sig { params(document: Document).void }
       def initialize(document)
         super
 
         @ranges = T.let([], T::Array[Interface::FoldingRange])
-        @partial_range = T.let(nil, T.nilable(PartialRange))
+        @requires = T.let([], T::Array[YARP::CallNode])
```

Comment:
Previously, we needed partial ranges for comments and requires. Now that comments are no longer nodes, we only need it for requires.

We basically accumulate require call nodes into this array until we find anything that isn't a require. Then we push a range for all of those require nodes.

Reply:
Just to confirm, if you had some requires separated by a comment, that would be two require ranges?

Reply:
Yes, that's right. If we find any nodes that aren't a require, we consider that to be the end of the require range and start a new one if necessary.